homeassistant/components/filter/sensor.py

```diff
@@ -231,6 +239,11 @@ def _update_filter_sensor_state_event(
         _LOGGER.debug("Update filter on event: %s", event)
         self._update_filter_sensor_state(event.data["new_state"])
 
+    def update(self) -> None:
+        """Update TimeSMAFilter value."""
+        temp_state = _State(dt_util.now(), 0)
+        self._run_filters(temp_state, timed_update=True)
```

Comment:
why not testing the states ? (why the need to split _update into _run ?)

Reply:
Mostly, there was a lot of redundancy in there for inserting a new state that didn't really apply to an update. I could be convinced to put it back though if you think the flow would be easier to follow.

Reply:
Ok. I merged it back to the way it was. It's easier to follow.